### ISSUE TITLE
Blank white window on NixOS + MNT Reform

### ISSUE THREAD

**Issue by jfrederickson** [atlas-engineer/nyxt#2258]
**Describe the bug**

I'm running NixOS 21.11 on my MNT Reform, and I've installed Nyxt from nixos-unstable to avoid #150412. However, upon starting nyxt, the application window is a blank white window, and I get errors in the console about not being able to create an EGL context (see logs below).

**Precise recipe to reproduce the issue**

Install NixOS 21.11 on MNT Reform. Install Nyxt from nixos-unstable using home-manager like so:

```
let
  pkgsUnstable = import <unstable> {};
in {
...
home.packages = with pkgs; [
    nheko
    gomuks
    tut
  ] ++ (with pkgsUnstable; [
    nyxt
  ]);
}
```

Start Nyxt from a console.

**Information**
- OS name+version: NixOS 21.11
- Graphics card and driver: GC7000Lite/etnaviv
- Desktop environment / Window manager name+version: Sway 1.7
- How you installed Nyxt (Guix pack, package manager, build from source): nix home-manager
- Information from `show-system-information`:

```
Nyxt version: 2.2.3
Renderer version: GTK
Operating system kernel: Linux 5.17.6
Lisp implementation: SBCL 2.2.4.nixos (Dynamic space size: 1073741824)
Features: (:WEBKIT2 :WEBKIT2-2.36 :WEBKIT2-CORS-ALLOWLIST :WEBKIT2-PASTE-PLAINTEXT
 :WEBKIT2-TRACKING :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA
 :WEBKIT2-SANDBOXING :GTK-3-22 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14
 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20
 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4
 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF :GLIB-2-30 :GLIB-2-32 :GLIB-2-34
 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48
 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56 :GLIB-2-58 :GLIB :NYXT-2
 :FSET-EXT-STRINGS :CUSTOM-HASH-TABLE-NATIVE :SWANK :PLUMP-UTF-32 :GLOBAL-VARS
 :DECLARE-TYPES :PARENSCRIPT :NAMED-READTABLES :LPARALLEL :21BIT-CHARS :CHUNGA
 :FLEXI-STREAMS :CLOSER-MOP :CL-PPCRE-UNICODE :CL-UNICODE :CL-PPCRE
 :BORDEAUX-THREADS :LPARALLEL.WITH-CLTL2 :LPARALLEL.WITH-CAS
 :LPARALLEL.WITH-STEALING-SCHEDULER :SPLIT-SEQUENCE CHIPZ-SYSTEM:GRAY-STREAMS
 CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:UNIX :CFFI CFFI-SYS::FLAT-NAMESPACE
 ALEXANDRIA::SEQUENCE-EMPTYP :FAST-IO-SV :FAST-IO :SBCL-USES-SB-ROTATE-BYTE
 :ASDF-SYSTEM-CONNECTIONS :CL-JSON-CLOS :CL-JSON :THREAD-SUPPORT :ASDF3.3
 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX :NON-BASE-CHARS-EXIST-P
 :ASDF-UNICODE :ARM64 :GENCGC :64-BIT :ANSI-CL :COMMON-LISP :ELF
 :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN :PACKAGE-LOCAL-NICKNAMES :SB-LDB
 :SB-PACKAGE-LOCKS :SB-THREAD :SB-UNICODE :SBCL :UNIX)

ASDF version: 3.3.4
ASDF registries: (NYXT-SOURCE-REGISTRY ENVIRONMENT-SOURCE-REGISTRY USER-SOURCE-REGISTRY
                      USER-SOURCE-REGISTRY-DIRECTORY
                      DEFAULT-USER-SOURCE-REGISTRY SYSTEM-SOURCE-REGISTRY
                      SYSTEM-SOURCE-REGISTRY-DIRECTORY
                      DEFAULT-SYSTEM-SOURCE-REGISTRY)
Critical dependencies: (/nix/store/hip5rlsfnsr2lm3i4si359370za9bp3d-lisp-cl-cffi-gtk-20201220-git/lib/common-lisp/cl-cffi-gtk/gtk/cl-cffi-gtk.asd
 /nix/store/kqri0zgd5qrma0dr2h1g7kzmwjim4dg9-lisp-cl-gobject-introspection-20210124-git/lib/common-lisp/cl-gobject-introspection/cl-gobject-introspection.asd
 /nix/store/cl8jdni1dyhnqs23gcxmh1bg9p31zdhx-lisp-cl-webkit2-cl-webkit-20211209-git/lib/common-lisp/cl-webkit2/webkit2/cl-webkit2.asd)
```

**Output when started from a shell**

```
[jfred@jfred-reform:~]$ nyxt 
<INFO> [17:23:09] Listening to socket "/run/user/1000/nyxt/nyxt.socket".

Nyxt version 2.2.3
Could not determine the accessibility bus address
EGLDisplay Initialization failed: EGL_NOT_INITIALIZED
EGLDisplay Initialization failed: EGL_NOT_INITIALIZED
EGLDisplay Initialization failed: EGL_NOT_INITIALIZED

** (nyxt:89868): WARNING **: 17:23:12.336: GDK is not able to create a GL context, falling back to glReadPixels (slow!): Unable to create a GL context
Cannot create EGL context: invalid display (last error: EGL_SUCCESS)

** (nyxt:89868): WARNING **: 17:23:12.340: GDK is not able to create a GL context, falling back to glReadPixels (slow!): Unable to create a GL context
Cannot create EGL context: invalid display (last error: EGL_SUCCESS)

** (nyxt:89868): WARNING **: 17:23:12.552: GDK is not able to create a GL context, falling back to glReadPixels (slow!): Unable to create a GL context
Cannot create EGL context: invalid display (last error: EGL_SUCCESS)
```

I gathered from some searching that for the warning in that output I should try running it with `GDK_GL=gles`, which seems to have done something, but doesn't change the white window:

```
[jfred@jfred-reform:~]$ GDK_GL=gles nyxt 
<INFO> [17:24:32] Listening to socket "/run/user/1000/nyxt/nyxt.socket".
Nyxt version 2.2.3
Could not determine the accessibility bus address
EGLDisplay Initialization failed: EGL_NOT_INITIALIZED
EGLDisplay Initialization failed: EGL_NOT_INITIALIZED
EGLDisplay Initialization failed: EGL_NOT_INITIALIZED
Cannot create EGL context: invalid display (last error: EGL_SUCCESS)
Cannot create EGL context: invalid display (last error: EGL_SUCCESS)
Cannot create EGL context: invalid display (last error: EGL_SUCCESS)
```

**Comment by Ambrevar** [comment]
Probably an issue with your graphics card.  Try setting the environment variable `WEBKIT_DISABLE_COMPOSITING_MODE` to `1` before starting Nyxt.

**Comment by jfrederickson** [comment]
Hmm. So that seems to have worked, at least initially, but then I ran into https://github.com/NixOS/nixpkgs/issues/158005 (and then [another issue](https://gist.github.com/jfrederickson/8d096c33e4f6637396810cc86042faa3) when I tried disabling sandboxing temporarily as a workaround as mentioned in that nixpkgs issue). Will have to dig a bit more info those.

In any case, I suppose the kernel I'm using may not have some patches required for GPU acceleration. I know they've gotten it working on the Librem 5 which uses the same GPU, but i wonder if they still have some out-of-tree patches needed for that...
